pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1211,8 +1211,10 @@ impl DownloadedLayer {
             // this will be a permanent failure
             .context("load layer");
 
-            if res.is_err() {
+            if let Err(e) = res.as_ref() {
                 LAYER_IMPL_METRICS.inc_permanent_loading_failures();
+                // TODO(#5815): we are not logging all errors, so temporarily log them here as well
+                tracing::error!("layer loading failed permanently: {e:#}");
```

Comment:
Maybe include the layer name in here in case the error doesn't?

Reply:
With [f57b140](https://github.com/neondatabase/neon/pull/5816/commits/f57b1406ed590b337a9a04d702716c7fcb5daf8b)

Reply:
Regarding the newlines I was (wrongly) thinking about the debug impl. Thanks for explaining.